src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -746,11 +756,17 @@ protected void handleVersionUpgrades(Connection conn) throws SQLException, Inval
 	}
 
 	private String getPositionSHA() {
-		return getSchemaPositionSHA(serverID, position.getFile(), position.getOffset());
+		return getSchemaPositionSHA(serverID, position);
 	}
 
-	public static String getSchemaPositionSHA(Long serverID, String binlogFilename, Long binlogPosition) {
-		String shaString = String.format("%d/%s/%d", serverID, binlogFilename, binlogPosition);
+	public static String getSchemaPositionSHA(Long serverID, Position position) {
+		BinlogPosition binlogPosition = position.getBinlogPosition();
+		String shaString = String.format("%d/%s/%d/%d",
+			serverID,
+			binlogPosition.getFile(),
+			binlogPosition.getOffset(),
+			position.getLastHeartbeatRead()
+		);
```

Comment:
This could break `findSchemaForPositionSHA` since the existing rows in `schemas` table have the old sha?